cryosparc2/protocols/protocol_base.py

```diff
@@ -190,10 +197,81 @@ def _importParticles(self):
 
         self.currenJob = pwobj.String(self.importedParticles.get())
         self._store(self)
-        self.par = pwobj.String(self.importedParticles.get() + '.imported_particles')
+        self.par = pwobj.String(self.importedParticles.get() +
+                                '.imported_particles')
 
     def setAborted(self):
         """ Set the status to aborted and updated the endTime. """
         pw.EMProtocol.setAborted(self)
         killJob(str(self.projectName.get()), str(self.currenJob.get()))
         clearJob(str(self.projectName.get()), str(self.currenJob.get()))
+
+    def createFSC(self, idd, imgSet, vol):
+        import requests
+        # Need to get the cryosparc master address
+        system_info = getSystemInfo()
+        status_errors = system_info[0]
+        if not status_errors:
+            system_info = eval(system_info[1])
+            master_hostname = system_info.get('master_hostname')
+            port_webapp = system_info.get('port_webapp')
+
+            url = "http://%s:%s/file/%s" % (master_hostname, port_webapp, idd)
+            fscRequest = requests.get(url, allow_redirects=True)
+            fscFile = "fsc.txt"
+            fscFilePath = os.path.join(self._getExtraPath(), fscFile)
+
+            # Convert into scipion fsc format
+            open(fscFilePath, 'wb').write(fscRequest.content)
+            f = open(fscFilePath, 'r')
+            lines = f.readlines()
+            wv = []
+            corr = []
+            for x in lines[1:-1]:
+                wv.append(str(float(x.split('\t')[0]) / (
+                        int(self._getInputParticles().getDim()[0]) * float(
```

Comment:
I think it's better to define this outside of the loop so they dont get called every time

Reply:
yes, it can be improved